packages/pluggableWidgets/accordion-web/src/components/Accordion.tsx

```diff
@@ -26,56 +26,101 @@ export function Accordion(props: AccordionProps): ReactElement | null {
+            parent={container}
+            id={`${props.id}AccordionGroup${index}`}
+            collapsible={props.collapsible}
+            collapsedAccordionGroupsDispatch={collapsedAccordionGroupsDispatch}
             {...group}
             collapsed={collapsedAccordionGroups[index]}
-            animateCollapsing={props.animateCollapsing}
-            generateIcon={props.generateHeaderIcon}
+            animateContent={props.animateContent}
+            generateHeaderIcon={props.generateHeaderIcon}
             showHeaderIcon={props.showGroupHeaderIcon}
         />
     ));
 
     return (
         <div
+            ref={updateContainer}
             id={props.id}
             className={classNames("widget-accordion", { "widget-accordion-preview": props.previewMode }, props.class)}
             style={props.style}
-            tabIndex={props.tabIndex}
+            data-focusindex={props.tabIndex || 0}
         >
             {accordionGroupElements}
         </div>
     );
 }
 
-interface AccordionGroupWrapperProps extends Omit<AccordionGroupProps, "toggleCollapsed"> {
+interface AccordionGroupWrapperProps extends Omit<AccordionGroupProps, "toggleCollapsed" | "changeFocus"> {
     index: number;
-    collapsedAccordionGroupsDispatch?: Dispatch<CollapsedAccordionGroupsReducerAction>;
+    parent?: HTMLDivElement;
+    collapsedAccordionGroupsDispatch: Dispatch<CollapsedAccordionGroupsReducerAction>;
 }
 
 function AccordionGroupWrapper(props: AccordionGroupWrapperProps): ReactElement {
+    const { collapsedAccordionGroupsDispatch } = props;
+
     const toggleCollapsedState = useCallback(() => {
         if (props.collapsed) {
-            props.collapsedAccordionGroupsDispatch!({ type: "expand", index: props.index });
+            collapsedAccordionGroupsDispatch({ type: "expand", index: props.index });
         } else {
-            props.collapsedAccordionGroupsDispatch!({ type: "collapse", index: props.index });
+            collapsedAccordionGroupsDispatch({ type: "collapse", index: props.index });
         }
-    }, [props.collapsed, props.collapsedAccordionGroupsDispatch, props.index]);
+    }, [props.collapsed, collapsedAccordionGroupsDispatch, props.index]);
+
+    const focusAccordionGroupHeaderElement = useCallback(
+        (focusedHeaderButton: EventTarget | null, focusTargetHeader: Target): void => {
+            if (props.parent && focusedHeaderButton && focusedHeaderButton instanceof Node) {
+                const headerButtons: HTMLDivElement[] = Array.from(
+                    props.parent.querySelectorAll(
+                        ":scope > .widget-accordion-group > .widget-accordion-group-header > .widget-accordion-group-header-button"
+                    )
+                );
+
+                switch (focusTargetHeader) {
+                    case Target.FIRST:
+                        headerButtons[0].focus();
```

Comment:
Will this be always defined?

Reply:
Hmm, not in the case that all accordion group are conditionally hidden, but that would need to happen in the time between pressing the key and shifting focus. To be sure, I can do a length check before.

Reply:
Fixed.